### ISSUE TITLE
Doc #inst and #uuid literals

### ISSUE THREAD

**Issue by puredanger** [clojure/clojure-site#120]
These aren't documented anywhere right now, would be nice to add maybe on the readers page.

Tiny bit in the changelog here:
https://github.com/clojure/clojure/blob/master/changes.md#211-instant-literals
https://github.com/clojure/clojure/blob/master/changes.md#212-uuid-literals

**Comment by daveliepmann** [comment]
Do you have verbiage in mind? I imagine adding text rather close to the changelog directly into *https://clojure.org/reference/reader#_tagged_literals* could suffice:

>Tagged literals are Clojure’s implementation of edn tagged elements. 
>
>For instance, Clojure supports literals for UUIDs in the form `#uuid "uuid-string"`, which is parsed as a `java.util.UUID`.
>
>...
>
>...Default reader tags are defined in [default-data-readers](https://clojure.github.io/clojure/clojure.core-api.html#clojure.core/default-data-readers) but may be overridden in `data_readers.clj` or by rebinding `*data-readers*`. For example, Clojure supports literals for instants in the form `#inst "yyyy-mm-ddThh:mm:ss.fff+hh:mm"`, which is parsed by default as a `java.util.Date`. These can be parsed as a `java.util.Calendar` or `java.util.Timestamp` by binding `*data-readers*` to use `clojure.instant/read-instant-calendar` or `clojure.instant/read-instant-timestamp`.
>
>If no data reader is found for a tag...

**Comment by puredanger** [comment]
While I would generally not encouraging copying content, the Clojure changelog is a very good place to steal from. :)  I think the other aspect to maybe think about for this one is that #uuid and #inst are actually edn features, not just Clojure ones. So referring to https://github.com/edn-format/edn/blob/master/README.md seems pretty essential.

**Comment by slipset** [comment]
Would this be solved by the weird characters?

**Comment by daveliepmann** [comment]
@slipset I don't see much overlap. [Reading Clojure Characters](https://clojure.org/guides/weird_characters) seems targeted at identifying them, providing a basic overview, and links to further reading. As I understand it, the issue here is aimed at documenting functionality, including how to change their parsing with `*data-readers*`.

@puredanger Regarding your last comment, the reader page already links to [edn-format/edn#tagged-elements](https://github.com/edn-format/edn#tagged-elements) in the first paragraph. What do you have in mind for further reference?

**Comment by puredanger** [comment]
Agreed, what is needed is more comprehensive reference material. edn defines the syntax for tagged elements but does not say anything about how readers for the tags are defined and managed. Clojure's solution to this is what needs to be documented. 

Regarding the link to edn, you're right so reference is covered. I still think it would be good to reinforce what tagged literal readers are provided by default, how data readers can be provided at startup, and the various features available for changing the data reader behavior at runtime (`*data-readers*` and `*default-data-reader-fn*`. 

Since the release notes were written, there have been a few notable additions as well. Namely, data_readers can now be either a .clj and .cljc file (for portability to cljs), and the new function support for generic tagged literals in the functions `tagged-literal` and `tagged-literal?`. You can combine all this together to install a default data reader fn that does not fail in the presence of an unknown tagged literal but actually instantiates a tagged literal instance that can carry the data and be passed on via printing to another system and/or programatically be inspected (tagged literal instances have ILookup support for `:tag` and `:form` keys). Some of this info can be found at https://dev.clojure.org/display/design/Reader+Conditionals.
